apps/dashboard/src/hooks/use-billing-subscription.ts

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { differenceInDays, isSameDay } from 'date-fns';
+import { ApiServiceLevelEnum, GetSubscriptionDto } from '@novu/shared';
+import { useAuth } from '@/context';
+import { getBillingSubscription } from '@/api/billing';
+import { QueryKeys } from '@/utils/query-keys';
+
+const today = new Date();
+
+export type UseSubscriptionType = GetSubscriptionDto & { trial: { daysLeft: number }; isLoading: boolean };
+
+export const useBillingSubscription = () => {
+  const { currentOrganization } = useAuth();
+
+  const { data: subscription, isLoading: isLoadingSubscription } = useQuery<GetSubscriptionDto>(
+    [QueryKeys.billingSubscription, currentOrganization?._id],
+    getBillingSubscription,
+    {
+      enabled: !!currentOrganization,
+      initialData: {
```

Comment:
Should we maybe set initial data to `null` and mark them as not available instead of a full JSON payload for a free subscription?

Usually, these defaults might result in a UI flicker, rendering stuff for `Free` and then, when fetching returns, rendering the actual data.